cmd/clusterctl/client/tree/discovery.go

```diff
@@ -151,21 +245,58 @@ func Discovery(ctx context.Context, c client.Client, namespace, name string, opt
 		}
 	}
 
-	// Handles orphan machines.
-	if len(machineMap) < len(machinesList.Items) {
-		other := VirtualObject(cluster.Namespace, "OtherGroup", "Other")
-		tree.Add(workers, other)
+	return nil
+}
 
-		for i := range machinesList.Items {
-			m := &machinesList.Items[i]
-			if _, ok := machineMap[m.Name]; ok {
-				continue
-			}
-			addMachineFunc(other, m)
+func addMachinePoolsToObjectTree(ctx context.Context, c client.Client, cluster *clusterv1.Cluster, workers *unstructured.Unstructured, machinePoolList *expv1.MachinePoolList, tree *ObjectTree, options DiscoverOptions) {
+	for i := range machinePoolList.Items {
+		mp := &machinePoolList.Items[i]
+		addOpts := make([]AddObjectOption, 0)
+		if !options.ShowTemplates {
+			addOpts = append(addOpts, GroupingObject(true))
+		}
+		tree.Add(workers, mp, addOpts...)
+
+		if options.ShowTemplates {
```

Comment:
We could probably add a "ShowMachinePools" flag similar to "ShowMachineSets" and replicate the behavior once MachinePoolMachines are added.

Reply:
Actually we might not need that at all since we have `--echo` and `--grouping=false` already